solutions/src/test/java/ru/sberbank/school/task10/ScalableThreadPoolTest.java

```diff
@@ -0,0 +1,115 @@
+package ru.sberbank.school.task10;
+
+import org.junit.jupiter.api.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ScalableThreadPoolTest {
+
+    @Test
+    public void incorrectThreadNumberTest() {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> new ScalableThreadPool(0, 4));
+        Assertions.assertThrows(IllegalArgumentException.class, () -> new ScalableThreadPool(3, 1));
+    }
+
+    @Test
+    public void interruptTest() throws InterruptedException {
+        Runnable task = () -> Assertions.assertThrows(InterruptedException.class, () -> Thread.sleep(10000));
+        ScalableThreadPool fixedThreadPool = new ScalableThreadPool(1, 2);
+        fixedThreadPool.start();
+        fixedThreadPool.execute(task);
+        Thread.sleep(1000);
+        fixedThreadPool.stopNow();
+    }
+
+    @Test
+    @DisplayName("Тест вызова метода execute до start")
+    public void testRunExecuteBeforeStart() {
+        ScalableThreadPool scalableThreadPool = new ScalableThreadPool(2,4);
+        Assertions.assertThrows(IllegalStateException.class, () -> scalableThreadPool.execute(() -> 1));
+    }
+
+    @Test
+    @DisplayName("Тест вызова метода stopNow до start")
+    public void testRunStopBeforeStart() {
+        ScalableThreadPool scalableThreadPool = new ScalableThreadPool(2,4);
+        Assertions.assertThrows(IllegalStateException.class, scalableThreadPool::stopNow);
+    }
+
+    @Test
+    public void testDoubleCallsStart() {
+        ScalableThreadPool scalableThreadPool = new ScalableThreadPool(2,4);
+        Assertions.assertThrows(IllegalStateException.class, () -> {
+            scalableThreadPool.start();
+            scalableThreadPool.start();
+        });
+        scalableThreadPool.stopNow();
+    }
+
+    @Test
+    public void threadCollectorTest() throws InterruptedException {
+        Runnable zeroThread = () -> {
+            try {
+                Thread.sleep(5000);
+            } catch (InterruptedException e) {
+
+            }
+        };
+        Runnable firstThread = () -> {
+            try {
+                Thread.sleep(10000);
+            } catch (InterruptedException e) {
```

Comment:
Empty catch block.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-2075441819>*